app/parse.py

```diff
@@ -8,8 +13,55 @@ class Quote:
     tags: list[str]
 
 
+QUOTE_FIELDS = [field.name for field in fields(Quote)]
+
+
+def parse_single_quote(quote_soup: BeautifulSoup) -> Quote:
+    return Quote(
+        text=quote_soup.select_one(".text").get_text(),
+        author=quote_soup.select_one(".author").get_text(),
+        tags=[
+            tag.get_text() for tag in quote_soup.select(".tag")
+        ]
+    )
+
+
+def get_single_page_quotes(page_soup: BeautifulSoup) -> list[Quote]:
+    quotes = page_soup.select(".quote")
+    return [parse_single_quote(quote_soup) for quote_soup in quotes]
+
+
+def get_quotes() -> list[Quote]:
+    all_quotes = []
+    base_url = "https://quotes.toscrape.com/"
+    current_url = base_url
+
+    while True:
+        page = requests.get(current_url).content
+        soup = BeautifulSoup(page, "html.parser")
+
+        all_quotes.extend(get_single_page_quotes(soup))
+
+        next_page = soup.find("li", class_="next")
+        if not next_page:
+            break
+
+        next_url = next_page.find("a")["href"]
+        current_url = base_url.rstrip("/") + next_url
```

Comment:
it is better to use `urllib.urljoin()` here

Reply:
agreed